src/Symfony/Component/Security/Core/User/ChainUserChecker.php

```diff
@@ -27,10 +29,16 @@ public function checkPreAuth(UserInterface $user): void
         }
     }
 
-    public function checkPostAuth(UserInterface $user): void
+    public function checkPostAuth(UserInterface $user /*, TokenInterface $token*/): void
     {
+        $token = 1 < \func_num_args() ? func_get_arg(1) : null;
+
         foreach ($this->checkers as $checker) {
-            $checker->checkPostAuth($user);
+            if ($token instanceof TokenInterface) {
```

Comment:
Do we really have to check if it's a `TokenInterface` or this is the `checker` work ? 
In this case we can do something like :

```php
$token = 1 < \func_num_args() ? func_get_arg(1) : false;
$checker->checkPostAuth($user, $token);
```

Reply:
I didn't want to pass `null` to `checkPostAuth()` in case someone implementing the new argument does only check the number of passed arguments but not the type.

Reply:
The problem is that if someone implements `checkPostAuth()` with this new second argument to prepare for the Symfony version where it is mandatory, it should expect to receive either nothing or a TokenInterface. Receiving anything else than a TokenInterface or nothing (while waiting the parameter is mandatory) would be really unexpected from the point of view of the person updating its implementation.

Reply:
Yes, I understand. In the future, the argument will be a `TokenInterface`, so that makes sense.